### PR TITLE
Add Ruby 3 and Mongo 5 to CI

### PR DESCRIPTION
**Pull request by CrazyOptimist**
- [x] Add Ruby 3.1.1, Mongo 5.0 to Mongoid test
- [x] Add Ruby 3.1.1 to ActiveRecord test
- [x] Modify database names to more meaningful ones
- [x] Fix Danger failure

### CODE REVIEW

**Comment by CrazyOptimist** [comment]
@dblock 
- Failed to figure out Ruby 3 with postgresql, guess it has something to do with passing yaml file differently according to ruby3 upgrade, but couldn't figure it out so far.
- Failed to fix Danger failure, please check the workflow file if I'm missing something.

**Comment by dblock** [comment]
> @dblock
> 
> * Failed to figure out Ruby 3 with postgresql, guess it has something to do with passing yaml file differently according to ruby3 upgrade, but couldn't figure it out so far.

The interface for `YAML.safe_load` is different in whatever ships with Ruby 3.1.1, copy-paste https://github.com/slack-ruby/slack-ruby-bot-server/blob/7623dcd9176ac8a85fb3154378baa298f45c40ac/spec/database_adapters/activerecord/activerecord.rb.

> * Failed to fix Danger failure, please check the workflow file if I'm missing something.

Not sure what its problem is, I gave it its own Gemfile like in slack-ruby-bot-server and it worked, https://github.com/slack-ruby/slack-ruby-bot-server-events/pull/15.

**Comment by CrazyOptimist** [comment]
>The interface for YAML.safe_load is different in whatever ships with Ruby 3.1.1, copy-paste https://github.com/slack-ruby/slack-ruby-bot-server/blob/7623dcd9176ac8a85fb3154378baa298f45c40ac/spec/database_adapters/activerecord/activerecord.rb.

I tried it already and failed. But let me try again.

**Comment by dblock** [comment]
> > The interface for YAML.safe_load is different in whatever ships with Ruby 3.1.1, copy-paste https://github.com/slack-ruby/slack-ruby-bot-server/blob/7623dcd9176ac8a85fb3154378baa298f45c40ac/spec/database_adapters/activerecord/activerecord.rb.
> 
> I tried it already and failed. But let me try again.

Show me a failure if you're still having trouble.

Add a CHANGELOG saying "Added support for Ruby 3.1." when both mongodb/postgresql work.

**Comment by CrazyOptimist** [comment]
```
An error occurred while loading ./spec/slack-ruby-bot-server-events/version_spec.rb.
Failure/Error: ActiveRecord::Tasks::DatabaseTasks.create(db_config)

ArgumentError:
  wrong number of arguments (given 3, expected 2)
```

**Comment by CrazyOptimist** [comment]
`db_config` is parsed correctly as I tried to print out:
```ruby
{"adapter"=>"postgresql", "pool"=>10, "timeout"=>5000, "encoding"=>"unicode", "database"=>"slack_ruby_bot_server_events_test", "url"=>"postgresql://test:password@localhost/slack_ruby_bot_server_events_test"}
```

**Comment by CrazyOptimist** [comment]
Something to do with this one maybe?  
https://www.ruby-lang.org/en/news/2019/12/12/separation-of-positional-and-keyword-arguments-in-ruby-3-0/

But curious I just copied it from `slack-ruby-bot-server` which passes

**Comment by dblock** [comment]
> Something to do with this one maybe? https://www.ruby-lang.org/en/news/2019/12/12/separation-of-positional-and-keyword-arguments-in-ruby-3-0/
> 
> But curious I just copied it from `slack-ruby-bot-server` which works without issues

If you look at the call stack of that error you'll notice that it's bringing in ActiveRecord 5 and errors like [so](https://stackoverflow.com/questions/69274304/rails-2-4-3-0-activerecord-add-modifier-wrong-number-of-arguments-given).

```
An error occurred while loading ./spec/slack-ruby-bot-server-events/api/endpoints/slack/actions_endpoint_spec.rb.
Failure/Error: ActiveRecord::Tasks::DatabaseTasks.create(db_config)

ArgumentError:
  wrong number of arguments (given 3, expected 2)
# /Users/dblock/.rvm/gems/ruby-3.1.1/gems/activerecord-5.0.7.2/lib/active_record/type/adapter_specific_registry.rb:7:in `add_modifier'
# /Users/dblock/.rvm/gems/ruby-3.1.1/gems/activerecord-5.0.7.2/lib/active_record/type.rb:22:in `add_modifier'
``` 

ActiveRecord 5 is just not compatible with this version of Ruby anymore. Once you move it up to 6, there are more problems that can be fixed the same way as slack-ruby-bot-server. Change this in Gemfile:

```
when 'activerecord' then
  gem 'activerecord', '~> 6.0.0'
  gem 'otr-activerecord'
  gem 'virtus'
  gem 'cursor_pagination', github: 'dblock/cursor_pagination', branch: 'misc' # rubocop:disable Bundler/OrderedGems
  gem 'pg'
```

**Comment by CrazyOptimist** [comment]
Thanks!

**Comment by CrazyOptimist** [comment]
Danger yells again this time, everything else is passing!
@dblock

**Comment by CrazyOptimist** [comment]
CI log doesn't give any clue for why danger failed :(

**Comment by dblock** [comment]
It does, see above: https://github.com/slack-ruby/slack-ruby-bot-server-events/pull/14#issuecomment-1075396697

**Comment by dblock** [comment]
@CrazyOptimist If you have time, take a look at https://github.com/slack-ruby/slack-ruby-bot-server/issues/148
